File: SEM_2/Programowanie Obiektowe/Lista_8/zadanie3.rb
class ONPExpression

  def initialize(expression)
    @expression = expression
  end

  def oblicz(variables)
    stack = []

    @expression.each do |element|
      case element
      when /\A-?\d+\z/ # integers
        stack.push(element.to_i)
      when "+", "-", "*", "/"
        right_operand = stack.pop
        left_operand = stack.pop
        result = calculate(left_operand, right_operand, element)
        stack.push(result)
      else # variable
        if variables[element]
          stack.push(variables[element])
        else
          raise "Nieznana zmienna: #{element}"
        end
      end

      puts stack.join(" ")
    end

    stack.pop
  end

  private

  def calculate(left, right, operator)
    case operator
    when "+"
      left + right
    when "-"
      left - right
    when "*"
      left * right
    when "/"
      left / right
    end
  end
end

expression = ["2", "3", "*", "x", "+", "5", "-"]
variables = { "x" => 4 }
onp = ONPExpression.new(expression)
puts "Wynik: #{onp.oblicz(variables)}"

File: Wybrane elementy praktyki projektowania oprogramowania/lista01/zadanie3.js

function is_prime(n){
    if(n === 0 || n === 1){
        return false;
    }
    else if(n === 2){
        return true;
    }
    else{
        for(let i = 2; i < n; i++){
            if(n % i === 0){
                debugger;
                return false;
            }
        }
        return true;
    }
}

const result = [];

for(let i = 0; i <= 100000; i++){
    if(is_prime(i)){
        result.push(i);
    }
}

console.log(result.join(", "));